src/structs.py

```diff
@@ -1126,6 +1126,7 @@ class GroundAtomsHoldResponse(Response):
 class DemonstrationQuery(Query):
     """A query requesting a demonstration to get from the state to a goal."""
     goal: Set[GroundAtom]
+    train_task_idx: int
```

Comment:
should we initialize the teacher with the training tasks instead, so that you only have to give `train_task_idx` here? i guess that's less explicit so maybe not?

Reply:
OTOH, it's slightly weird to give both of these fields...

Reply:
yeah I think the weirdness of having both fields outweighs the explicitness concern. I'll try it that way